src/positions/saga.ts

```diff
@@ -188,6 +201,65 @@ export function* handleEnableHooksPreviewDeepLink(
   }
 }
 
+export function* triggerShortcutSaga({ payload }: ReturnType<typeof triggerShortcut>) {
+  Logger.debug(`${TAG}/triggerShortcutSaga`, 'Initiating request to claim reward', payload)
+
+  const hooksApiUrl = yield select(hooksApiUrlSelector)
+
+  try {
+    const response = yield call(fetchWithTimeout, `${hooksApiUrl}/triggerShortcut`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    })
+    if (!response.ok) {
+      throw new Error(`Unable to trigger shortcut: ${response.status} ${response.statusText}`)
+    }
+
+    const { data } = yield call([response, 'json'])
+
+    const kit: ContractKit = yield call(getContractKit)
+    const walletAddress: string = yield call(getConnectedUnlockedAccount)
+    const normalizer = new TxParamsNormalizer(kit.connection)
+
+    Logger.debug(`${TAG}/triggerShortcutSaga`, 'Starting to claim reward(s)', data.transactions)
+
+    for (const transaction of data.transactions) {
+      applyChainIdWorkaround(transaction, yield call([kit.connection, 'chainId']))
+      const tx: CeloTx = yield call([normalizer, 'populate'], transaction)
+      const txo = buildTxo(kit, tx)
+
+      const receipt: CeloTxReceipt = yield call(
+        sendTransaction,
+        txo,
+        walletAddress,
+        newTransactionContext(TAG, 'Trigger shortcut')
+      )
```

Comment:
I think If there were really multiple transactions to send, this would wait unnecessarily for each of them to be included in different blocks. Instead of sending them all in parallel.

We don't need to address this now, just noting. Maybe we could add a TODO about this?

Reply:
i always get confused about this topic, won't we run into nonce issues if we try to send multiple transactions at the same time? i added a TODO for now, but happy to address in a follow up